predicthq/endpoints/v1/beam/endpoint.py

```diff
@@ -0,0 +1,235 @@
+from predicthq.endpoints.base import BaseEndpoint
+from .schemas import (
+    CreateAnalysisResponse,
+    AnalysisResultSet,
+    Analysis,
+    EventResultSet,
+    Event,
+    FeatureImportance,
+    ValueQuant,
+    CorrelationResultSet,
+)
+from predicthq.endpoints.decorators import accepts, returns
+from typing import overload, List
+from datetime import date
+
+
+class BeamEndpoint:
+    def __init__(self, client):
+        self.analysis = self.Analysis(client)
+
+    class Analysis(BaseEndpoint):
+        @overload
+        def create(
+            self,
+            name: str,
+            location__geopoint: dict,
+            location__radius: float = None,
```

Comment:
I think you need to mark these as `Optional` here and in all other params that have a default of `None`:
```suggestion
            location__radius: Optional[float] = None,
```

Reply:
🙌 Will update - thanks for catching that